tests/ert3/engine/integration/integration_utils.py

```diff
@@ -25,14 +26,22 @@ def assert_input_records(config, export_data):
                 parameter = p
                 break
 
-        input_records[record] = parameter.variables
+        assert parameter.variables or parameter.size
```

Comment:
Since there is a branching here dictating both the content of `input_records[record]` and the assertions done within the loop it might be easier to read and connect if you introduced a bool for this branching immediately after the assert.

```python
has_variables = parameter.variables is not None
```

and then use `has_variables` both when setting `input_records[record]` and when branching the asserts within the loop.

Reply:
Added a `has_variables` for readability. However, each `input_records[record]` need to be checked again for having `size` or `variables` before the asserts in the loop as the setup is now.